galsim/utilities.py

```diff
@@ -705,13 +705,12 @@ def horner2d(x, y, coefs, dtype=None, triangle=False):
         y = np.ascontiguousarray(y, dtype=float)
         coefs = np.ascontiguousarray(coefs, dtype=float)
     else:
-        x = np.array(x, copy=False, dtype=float)
-        y = np.array(y, copy=False, dtype=float)
-        coefs = np.array(coefs, copy=False, dtype=float)
+        x = np.array(x, copy=False)
```

Comment:
I think this change (and the 1d version too) makes the docstrings of `_horner`, `_horner2d` lies.  The dtype can now be complex.  Actually, looks like the docstrings were previous lies too.

Reply:
Also, s/coeff/coef for 1d and s/coeff/coefs for 2d.